teamscale_precommit_client/git_utils.py

```diff
@@ -22,16 +23,19 @@ def get_current_branch(path_to_repository):
         Returns:
             str: The current branch in the provided repository.
     """
-    repo = Repository(path_to_repository)
-    head = repo.lookup_reference("HEAD").resolve()
-    return head.shorthand
+    repo = Repo(path_to_repository)
+    return repo.active_branch.name
+
 
 def get_repo_root_from_file_in_repo(path_to_file_in_repo):
     """Get the repository root for the given path in the repository."""
-    git_folder = discover_repository(path_to_file_in_repo)
-    if not git_folder:
+    try:
+        repo = Repo(path=path_to_file_in_repo, search_parent_directories=True)
+        git_root = repo.git.rev_parse("--show-toplevel")
+        return git_root
+    except InvalidGitRepositoryError:
```

Comment:
Please add logging here for this error so users can understand what went wrong

Reply:
Already happens [here](https://github.com/cqse/teamscale-cli/blob/cr/23333_gitpython_lib/teamscale_precommit_client/precommit_client.py#L88). Just kept the API stable. With pygit, [discover_repository](https://www.pygit2.org/repository.html#pygit2.discover_repository) returned `None` instead of throwing an exception.